component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthenticator.java

```diff
@@ -0,0 +1,210 @@
+
+        SignedJWT signedJWT = null;
+        String jwtSubject;
+        try {
+            List oauthJWTAssertion = content.get(OAUTH_JWT_ASSERTION);
+            if (oauthJWTAssertion != null && oauthJWTAssertion != null) {
+                signedJWT = getSignedJWT(oauthJWTAssertion.toString());
+            }
+            ReadOnlyJWTClaimsSet claimsSet = jwtValidator.getClaimSet(signedJWT);
+            jwtSubject = jwtValidator.resolveSubject(claimsSet);
+
+        } catch (IdentityOAuth2Exception e) {
+            String message = "Error while obtaining the client id from the token request.";
+            throw new OAuthClientAuthnException(message, OAuth2ErrorCodes.INVALID_CLIENT);
+        }
+        return jwtSubject;
+    }
+
+    private SignedJWT getSignedJWT(String assertion) throws OAuthClientAuthnException {
+
+        SignedJWT signedJWT;
+        if (isEmpty(assertion)) {
+            String errorMessage = "No Valid Assertion was found for " + Constants.OAUTH_JWT_BEARER_GRANT_TYPE;
+            throw new OAuthClientAuthnException(errorMessage);
+        }
+        try {
+            signedJWT = SignedJWT.parse(assertion);
+            logJWT(signedJWT);
+        } catch (ParseException e) {
+            String errorMessage = "Error while parsing the JWT.";
+            throw new OAuthClientAuthnException(errorMessage);
+        }
+        if (signedJWT == null) {
+            String errorMessage = "No Valid Assertion was found for " + OAUTH_JWT_BEARER_GRANT_TYPE;
+            throw new OAuthClientAuthnException(errorMessage, OAuth2ErrorCodes.INVALID_REQUEST);
+        }
+        return signedJWT;
+    }
+
+    private void logJWT(SignedJWT signedJWT) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("JWT Header: " + signedJWT.getHeader().toJSONObject().toString());
+            log.debug("JWT Payload: " + signedJWT.getPayload().toJSONObject().toString());
+
+            if (signedJWT.getSignature() != null) {
+                log.debug("Signature: " + signedJWT.getSignature().toString());
+            }
+        }
+    }
+
+    private boolean isValidJWTClientAssertionRequest(String clientAssertionType, String clientAssertion) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Authenticate Requested with : " + clientAssertionType + " and " +
+                    "private_key_jwt: " + clientAssertion);
+        }
+        return OAUTH_JWT_BEARER_GRANT_TYPE.equals(clientAssertionType) && isNotEmpty(clientAssertion);
+    }
+
+    private JWTValidator createJWTValidator(String tokenEPAlias, boolean preventJTIReplay) {
+
+        boolean preventTokenReuse = preventJTIReplay;
+        List<Object> mandatoryClaims = new ArrayList<>();
+
+        int notAcceptBeforeTimeInMins = DEFAULT_VALIDITY_PERIOD_IN_MINUTES;
+
+        try {
+            String rejectBeforePeriodConfigVal = String.valueOf(notAcceptBeforeTimeInMins);
```

Comment:
Shouldn't read the property from somewhere else rather using the variable `notAcceptBeforeTimeInMins`?

Reply:
notAcceptBeforeTimeInMins -rename this and check here